### PR TITLE
fix: add python version determination

### PR DESCRIPTION
**Pull request by NikoKrismerInnio**

### CODE REVIEW

File: workflow-templates/prefect-dependency-check.yml [hunk 1]
```diff
@@ -20,14 +20,50 @@ on:
...
+
+      - name: Check python version determination (.python-version)
+        if: steps.check_file_pyproject.outputs.files_exists == 'false'
+        id: check_file_python_version
+        uses: andstor/file-existence-action@v3
+        with:
+          fail: false
+          files: '.python-version'
+
+      - name: Use pyproject.toml as python version file
```
Comment (lukasthalerINNIO) [inline workflow-templates/prefect-dependency-check.yml:48]:
```suggestion
      - name: Use .python-version as python version file
```

File: workflow-templates/python-dependency-check.yml [hunk 2]
```diff
@@ -20,14 +20,50 @@ on:
...
+
+      - name: Check python version determination (.python-version)
+        if: steps.check_file_pyproject.outputs.files_exists == 'false'
+        id: check_file_python_version
+        uses: andstor/file-existence-action@v3
+        with:
+          fail: false
+          files: '.python-version'
+
+      - name: Use pyproject.toml as python version file
```
Comment (lukasthalerINNIO) [inline workflow-templates/python-dependency-check.yml:48]:
```suggestion
      - name: Use .python-version as python version file
```

File: workflow-templates/python-deploy.yml [hunk 3]
```diff
@@ -21,17 +21,53 @@ on:
...
+
+      - name: Check python version determination (.python-version)
+        if: steps.check_file_pyproject.outputs.files_exists == 'false'
+        id: check_file_python_version
+        uses: andstor/file-existence-action@v3
+        with:
+          fail: false
+          files: '.python-version'
+
+      - name: Use pyproject.toml as python version file
```
Comment (lukasthalerINNIO) [inline workflow-templates/python-deploy.yml:49]:
```suggestion
      - name: Use .python-version as python version file
```

File: workflow-templates/python-test.yml [hunk 4]
```diff
@@ -13,12 +13,49 @@ on:
...
+
+      - name: Check python version determination (.python-version)
+        if: steps.check_file_pyproject.outputs.files_exists == 'false'
+        id: check_file_python_version
+        uses: andstor/file-existence-action@v3
+        with:
+          fail: false
+          files: '.python-version'
+
+      - name: Use pyproject.toml as python version file
```
Comment (lukasthalerINNIO) [inline workflow-templates/python-test.yml:41]:
```suggestion
      - name: Use .python-version as python version file
```

**Review by lukasthalerINNIO — CHANGES_REQUESTED** [review]
There is one mis-named step that copy-pasted its way through all actions and one block that I think is redundant

File: workflow-templates/python-dependency-check.yml [hunk 5]
```diff
@@ -52,6 +88,27 @@ jobs:
...
+        with:
+          fail: false
+          files: '.python-version'
+
+      - name: Check python version determination
+        if: steps.check_file_pyproject.outputs.files_exists == 'false' &&  steps.check_file_python_version.outputs.files_exists == 'false'
+        run: |
+          echo "::error Either pyproject.toml or .python-version must exists (and contain a python version number)"
+          exit 1
+
```
[the remark this answers is not in the document]
Comment (NikoKrismerInnio) [reply workflow-templates/python-dependency-check.yml:111]: true... another copy (instead of cut) mistake. thanks for finding it :)

**Review by lukasthalerINNIO — APPROVED** [review]
lgtm

**Pull request merged**